src/lib/server/websearch/sentenceSimilarity.ts

```diff
@@ -0,0 +1,39 @@
+import type { Tensor } from "@xenova/transformers";
+import { pipeline, dot } from "@xenova/transformers";
+
+// see here: https://github.com/nmslib/hnswlib/blob/359b2ba87358224963986f709e593d799064ace6/README.md?plain=1#L34
+function innerProduct(tensor1: Tensor, tensor2: Tensor) {
+	return 1.0 - dot(tensor1.data, tensor2.data);
```

Comment:
ah nice dot product is already implemented in there 🎉

Reply:
yes, dot product - the operation of the brain